models/context.py

```diff
@@ -23,10 +24,68 @@ class ResourceObservation(_BaseObs):
     type: Literal[ObservationType.RESOURCE]
     energy_yield: int
     available: bool
+    required_agents: int
+    harvesting_area: int
+    mining_time: int
+    being_harvested: bool
+    num_harvester: int
 
     def __str__(self) -> str:
-        availability = "available" if self.available else "unavailable"
-        return f"[ID: {self.id}; type: {self.type.value}; location: {self.location}; distance: {self.distance}; energy yield: {self.energy_yield}; {availability}]"
+
+        obs = f"""A resource at location {self.location} ({self.distance} units away) with
+                  energy yield {self.energy_yield} and mining time {self.mining_time}."""
```

Comment:
please keep new changes I made in #51 to the prompting style
```suggestion
availability = "available" if self.available else "unavailable"
obs = f"[ID: {self.id}; type: {self.type.value}; location: {self.location}; distance: {self.distance}; energy yield: {self.energy_yield}; mining time: {self.mining_time}; {availability}]"
```